api/v1alpha1/nodepool_types.go

```diff
@@ -419,11 +420,86 @@ type PowerVSNodePoolPlatform struct {
 	ImageDeletePolicy string `json:"imageDeletePolicy,omitempty"`
 }
 
+// KubevirtCompute contains values associated with the virtual compute hardware requested for the VM.
+type KubevirtCompute struct {
+	// Memory represents how much guest memory the VM should have
+	//
+	// +optional
+	// +kubebuilder:default="4Gi"
+	Memory *resource.Quantity `json:"memory"`
+
+	// Cores represents how many cores the guest VM should have
+	//
+	// +optional
+	// +kubebuilder:default=2
+	Cores *uint32 `json:"cores"`
+}
+
+// KubevirtPersistentVolume containes the values involved with provisioning persistent storage for a KubeVirt VM.
+type KubevirtPersistentVolume struct {
+	// Size is the size of the persistent storage volume
+	//
+	// +optional
+	// +kubebuilder:default="16Gi"
+	Size *resource.Quantity `json:"size"`
+	// StorageClass is the storageClass used for the underlying PVC that hosts the volume
+	//
+	// +optional
+	StorageClass *string `json:"storageClass,omitempty"`
+}
+
+// KubevirtRootVolume represents the volume that the rhcos disk will be stored and run from.
+type KubevirtRootVolume struct {
+	// Image represents what rhcos image to use for the node pool
+	Image *KubevirtDiskImage `json:"diskImage"`
+
+	// KubevirtVolume represents of type of storage to run the image on
+	KubevirtVolume `json:",inline"`
+}
+
+// KubevirtVolumeType is a specific supported KubeVirt volumes
+//
+// +kubebuilder:validation:Enum=Persistent
+type KubevirtVolumeType string
+
+const (
+	// KubevirtVolumeTypePersistent represents persistent volume for kubevirt VMs
+	KubevirtVolumeTypePersistent KubevirtVolumeType = "Persistent"
+)
+
+// KubevirtVolume represents what kind of storage to use for a KubeVirt VM volume
+type KubevirtVolume struct {
+	// Type represents the type of storage to associate with the kubevirt VMs.
+	//
+	// +unionDiscriminator
+	// +kubebuilder:default=Persistent
+	Type KubevirtVolumeType `json:"type"`
```

Comment:
this also applies here https://github.com/openshift/hypershift/pull/1314/files#r865699495